src/frontend/components/JoinDiscussionAskButton/index.tsx

```diff
@@ -28,18 +29,46 @@ const messages = defineMessages({
 });
 
 export const JoinDiscussionAskButton = () => {
-  const { asked, accepted, rejected, setAsked } = useParticipantWorkflow(
-    (state) => ({
-      asked: state.asked,
-      accepted: state.accepted,
-      rejected: state.rejected,
-      setAsked: state.setAsked,
-    }),
-  );
+  const [username, setUsername] = useState('');
+  const [showUsernameForm, setShowUsernameForm] = useState(false);
+  const {
+    asked,
+    accepted,
+    rejected,
+    usernameAlreadyExisting,
+    reset,
+    setAsked,
+  } = useParticipantWorkflow((state) => ({
+    asked: state.asked,
+    accepted: state.accepted,
+    rejected: state.rejected,
+    usernameAlreadyExisting: state.usernameAlreadyExisting,
+    setAsked: state.setAsked,
+    reset: state.reset,
+  }));
+
+  useEffect(() => {
+    if (usernameAlreadyExisting) {
+      setShowUsernameForm(true);
+    }
+  }, [usernameAlreadyExisting]);
```

Comment:
I wonder we could not remove this. No quick solution in mind yet.

Reply:
We have to show the form if after a connection we have an error response because the username is already used.

Reply:
I wondered if we cannot remove the `showUsernameForm` state then just decides to show the form according to `username` and `usernameAlreadyExisting` values but to achieve that we have to manage form validation differently and it will complexify the code.